js/classes/entities/midground/weapon/projectile_base.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Class which provides basic functionality for enemy entities, including movement,
+ * taking damage, and block collisions.
+ * 
+ * @author Nathan Hinthorne
+ * @author Trae Claar
+ */
+class ProjectileBase {
+    /**
+     * Constructor for a ProjectileBase.
+     * 
+     * @param {Entity} projectile the projectile associated with this projectileBase
+     * @param {Vector} startPos the initial position of the projectile
+     * @param {Vector} targetPos the target position of the projectile
+     * @param {number} scaledSize the size of the projectile on the CANVAS
+     * @param {number} speed the speed of the projectile
+     * @param {function} onBlockCollision callback function invoked when the projectile collides with a block
+     * @param {function} onEnemyCollision callback function invoked when the projectile collides with an enemy
+     */
+    constructor(projectile, startPos, targetPos, scaledSize, speed, weight, onBlockCollision, onEnemyCollision) {
+        this.projectile = projectile;
+        projectile.pos = startPos;
+        projectile.targetPos = targetPos;
+        
+        projectile.action = "firing"; // or "idle" to start with?
+        projectile.scaledSize = scaledSize;
+        projectile.speed = speed;
+        projectile.weight = weight;
+        projectile.yVelocity = 0;
+
+        projectile.applyGravity = true; // used for bouncing projectiles
+        projectile.hasHit = false;
+
+        projectile.boundingBox = new BoundingBox(startPos, scaledSize);
+        projectile.lastBoundingBox = this.projectile.boundingBox;
+
+        this.onBlockCollision = onBlockCollision;
+        this.onEnemyCollision = onEnemyCollision;
+
+        // determine what direction to travel in
+        projectile.dir = Vector.direction(startPos, targetPos);
+    }
+
+    /**
+     * Updates the projectile's position and checks for collisions.
+     */
+    update() {
+        // move the projectile
+        if (!this.projectile.applyGravity) {
+            this.projectile.yVelocity = 0;
+        } else {
+            this.projectile.yVelocity += PHYSICS.GRAVITY_ACC * GAME.clockTick * this.projectile.weight;
+
+            // max out all downwards speed at the terminal velocity
```

Comment:
Why is this commented out? Also some comments I have seen such as on line 54 and 48 are redundant as the code is clear.